odoo/models.py

```diff
@@ -2650,6 +2650,10 @@ def _add_sql_constraints(self):
 
         for (key, definition, message) in self._sql_constraints:
             conname = '%s_%s' % (self._table, key)
+            if len(conname) > 63:
+                # TODO: remove before merge!
+                _logger.warning("Constraint name %r has more than 63 characters", conname)
```

Comment:
> FWIW there's a check_pg_name utility function to check for identifier names.

Yes I tried to use it but then I couldn't reconcile trying to (manually) chop the names with the possible prefixes and suffixes the constraint could get. `make_identifier` looks like the right solution, perhaps for master instead? Some standard constraints are already too long, e.g in 14.0 `mail_message_res_partner_needaction_rel_notification_partner_required`.

Regarding the warning. I could switch it to an info instead, since the log line is still useful, _wdyt?_

Reply:
@xmo-odoo your proposal (`make_identifier`) is the way to go for master, IMHO.
But for stable versions we need something more conservative, I believe.